src/blueapi/worker/task_worker.py

```diff
@@ -341,10 +412,26 @@ def _report_status(
 
     def _on_document(self, name: str, document: Mapping[str, Any]) -> None:
         if self._current is not None:
-            correlation_id = self._current.task_id
-            self._data_events.publish(
-                DataEvent(name=name, doc=document), correlation_id
-            )
+            with TRACER.start_as_current_span(
+                "_on_document",
+                context=self._context_register[self._current.task_id],
+                kind=SpanKind.PRODUCER,
+            ):
+                """Start a new span but inject the context cached when the current task
```

Comment:
Because, in the general case you could have documents arriving from different runs? Mainly however this makes it easier to spot which document, if any, causes a problem in the event of an error. Despite having distinct Spans, they will all be grouped as part of the same Trace (because of the context propagation) so you can see that  they all originated from the same call, whilst having them arranged vertically in sequence in the Jaeger UI.

Reply:
I see, I guess I imagined they would all be grouped by run or something, but as long as this doesn't cause too much spam in Jaeger that's fine